src/common/EmployerRecord.ts

```diff
@@ -8,61 +7,87 @@ export class EmployerRecord extends EmployerRecordBase {
 
 	public citations: Citation[] = [];
 
-	public static calculateRating(
-		citationCount: number,
-		score: number,
-	): EmployerRating {
-		if (citationCount === 0) {
-			return "fair";
-		}
-
-		const normalizedScore: number = citationCount / 2;
-
-		if (score < normalizedScore && score > -normalizedScore) {
-			return "fair";
-		}
-
-		return score > 0 ? "good" : "poor";
-	}
-
 	public static toMetadata(
 		original: EmployerRecord,
 		employerReferenceMap?: { [key: string]: EmployerRecord },
 	): EmployerRecordMetadata {
-		let citationCount: number = 0;
-		let score: number = 0;
-		let positives: number = 0;
-		let negatives: number = 0;
+		let citationRatings: number[] = [];
 
-		const updateRatings = (e?: EmployerRecord): void => {
-			if (!e) {
+		const addCitations = (e?: EmployerRecord): void => {
+			if (!e?.citations) {
 				return;
 			}
 
-			for (const citation of e.citations) {
-				citationCount++;
-				score += citation.positivity;
-
-				if (citation.positivity > 0) {
-					positives++;
-				} else if (citation.positivity < 0) {
-					negatives++;
-				}
-			}
+			e.citations.forEach((c: Citation) => {
+				citationRatings.push(c.positivity);
+			});
 		};
 
-		updateRatings(original);
+		addCitations(original);
 
 		if (employerReferenceMap) {
 			if (original.parentId) {
-				updateRatings(employerReferenceMap[original.parentId]);
+				addCitations(employerReferenceMap[original.parentId]);
+			}
+
+			original.childIds.forEach((childId: string) => addCitations(employerReferenceMap[childId]));
+		}
+
+		citationRatings = citationRatings.sort();
+
+		// Tweaking this to be larger than zero results in a broader range for "fair".
+		let normalizedScore: number = 0;
+
+		let negatives: number = 0;
+		let positives: number = 0;
+		let score: number = 0;
+
+		if (citationRatings.length === 1) {
+			score = citationRatings[0];
+		} else if (citationRatings.length > 1) {
+			normalizedScore = 1;
+
+			let i: number = 0;
```

Comment:
Would it be possible to rename this variable to make it more readable?

Reply:
I've reformatted the algo completely so this is gone.